extensions/purchase-label/functions/src/types.ts

```diff
@@ -1,6 +1,9 @@
+import { CreateLabelFromRateTypes } from 'shipengine/esm/create-label-from-rate';
 import { CreateLabelFromShipmentDetailsTypes } from 'shipengine/esm/create-label-from-shipment-details';
 
-export type RequestPayload = CreateLabelFromShipmentDetailsTypes.Params;
-export type ResponsePayload = CreateLabelFromShipmentDetailsTypes.Result;
+export type RequestPayload = CreateLabelFromRateTypes.Params &
+  CreateLabelFromShipmentDetailsTypes.Params;
+export type ResponsePayload = CreateLabelFromRateTypes.Result &
+  CreateLabelFromShipmentDetailsTypes.Result;
```

Comment:
Couldn't get this to work using TypeScript union, so I just went ahead with the intersection.

Reply:
An intersection is the correct way to type this. We want the resulting type to include all elements of the included types.